src/Core/src/Platform/Android/ComponentCallbacks.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using Android.App;
+using Android.Content;
+using Android.Content.Res;
+using Microsoft.Maui.LifecycleEvents;
+
+namespace Microsoft.Maui
+{
+	class ApplicationComponentCallbacks : ComponentCallbacks
+	{
+		Application application;
+
+		public ApplicationComponentCallbacks(Application application, IServiceProvider services)
+			: base(services) => this.application = application;
+
+		public override void OnConfigurationChanged(Configuration newConfig)
+			=> services.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationConfigurationChanged>(del => del(application, newConfig));
+
+		public override void OnLowMemory()
+			=> services.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationLowMemory>(del => del(application));
+
+		public override void OnTrimMemory(TrimMemory level)
+			=> services.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationTrimMemory>(del => del(application, level));
+	}
+
+	class ActivityComponentCallbacks : ComponentCallbacks
+	{
+		WeakReference<Activity> activity;
+
+		public ActivityComponentCallbacks(Activity activity, IServiceProvider services)
+			: base(services) => this.activity = new WeakReference<Activity>(activity);
+
+		public override void OnConfigurationChanged(Configuration newConfig)
+		{
+			if (!activity.TryGetTarget(out var target))
+				return;
+			MauiApplication.Current?.Application.ThemeChanged();
+			services.InvokeLifecycleEvents<AndroidLifecycle.OnConfigurationChanged>(del => del(target, newConfig));
+		}
```

Comment:
This is certainly a reason to make this PR land as we can move more out of the overrides in the maui activity as native embedding is not using that.